scalanet/src/io/iohk/scalanet/codec/EitherCodec.scala

```diff
@@ -0,0 +1,51 @@
+package io.iohk.scalanet.codec
+
+import java.nio.ByteBuffer
+
+import scodec.Attempt.{Failure, Successful}
+import scodec.bits.BitVector
+import scodec.{Attempt, Codec, DecodeResult, Err, SizeBound}
+
+class EitherCodec[A, B](implicit a: Codec[A], b: Codec[B], byteCodec: Codec[Byte]) extends Codec[Either[A, B]] {
```

Comment:
scodec has a special combinator called `either` in `scodec.codecs.either` which is exactly what you want